libusbcommunication/src/main/java/me/jahnen/libaums/libusbcommunication/LibusbCommunication.kt

```diff
@@ -87,13 +87,18 @@ class LibusbCommunication(
         // if LIBUSB_ERROR_NOT_FOUND might need reenumeration
         Log.d(TAG, "libusb reset returned $ret")
 
-        if (!deviceConnection!!.claimInterface(usbInterface, true)) {
-            throw IOException("Could not claim interface, errno: ${ErrNo.errno} ${ErrNo.errstr}")
+        var counter = 3
+        while(!deviceConnection!!.claimInterface(usbInterface, true) && counter >= 0) {
+            if (counter == 0) {
+                throw IOException("Could not claim interface, errno: ${ErrNo.errno} ${ErrNo.errstr}")
+            }
+            Thread.sleep(800)
+            counter--
         }
     }
 
     override fun clearFeatureHalt(endpoint: UsbEndpoint) {
-        val ret = nativeClearHalt(libUsbHandle, usbInterface.id)
+        val ret = nativeClearHalt(libUsbHandle, endpoint.address)
```

Comment:
@Depau

Reply:
Oh I didn't see this. I'm not sure what's the matter though

Reply:
Well this never really worked, because the nativeClearHalt needs the endpoint address and not the interface id. This resulted in an error in the logs. But more of an fyi and not something which will help us solve the overall problem.